app/models/manageiq/providers/nuage/network_manager/event_catcher/stream.rb

```diff
@@ -2,17 +2,18 @@ class ManageIQ::Providers::Nuage::NetworkManager::EventCatcher::Stream
   include Vmdb::Logging
 
   def self.test_amqp_connection(options = {})
+    return false if options[:urls].empty?
     # Ensure we just test the connection. AMQP channel will be established and
     # started, however it will be immediately stopped.
     options[:test_connection] = true
-    begin
-      stream = new(options)
+
+    stream = new(options)
+    ok = stream.with_fallback_urls(options[:urls]) do
       stream.connection.run
-      true
-    rescue => e
-      _log.info("#{log_prefix} Failed connecting to ActiveMQ: #{e.message}")
-      raise
+      return true
     end
+    return ok if ok
```

Comment:
This looks interesting 😄. I'd make it either a bit more explicit, i.e. `return true if ok` or change the order, i.e.

```
raise MiqException::MiqHostError, "Could not ..." unless ok
return true
```

Reply:
Wow, did I really code this? 😄  Fixed.